src/piton/labelers/core.py

```diff
@@ -0,0 +1,381 @@
+from __future__ import annotations
+
+import pprint
+import collections
+import datetime
+import json
+import pickle
+import os
+from collections.abc import MutableMapping
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import (
+    Any,
+    DefaultDict,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
+
+from .. import Event, Patient
+from ..extension import datasets as extension_datasets
+
+import numpy as np
+
+@dataclass
+class TimeHorizon:
+    start: datetime.timedelta
+    end: datetime.timedelta
+
+@dataclass
+class SurvivalValue:
+    event_time: int # TODO - rename
+    is_censored: bool # TRUE if this patient was censored
+
+LabelType = Union[
+    Literal["boolean"],
+    Literal["numeric"],
+    Literal["survival"],
+    Literal["categorical"],
+]
+
+VALID_LABEL_TYPES = ["boolean", "numeric", "survival", "categorical"]
+
+
+# Make sure this is pickleable
+class Label:
```

Comment:
This should be implemented using the dataclasses supertype

@dataclasses.class
class Label(frozen=True):

This will automatically implement most of the other methods here and add a bunch of neat functionality.

Reply:
Not fixed?